### PR TITLE
Inject TagIterator into BlockIterator

### LINKED ISSUE #38: Inject TagIterator Into BlockIterator
### Housekeeping

- [X] I am a maintainer of dbt-common

### Short description

For greater flexibility, we need to inject the instance of TagIterator used for parsing into BlockIterator upon construction.

### Acceptance criteria

The 'data' parameter for BlockIterator has been replaced by a TagIterator instance.

### Suggested Tests

Existing tests will suffice.

### Impact to Other Teams

None

### Will backports be required?

No

### PR DESCRIPTION
**Pull request by peterallenwebb**
resolves #38

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dbt_common/clients/_jinja_blocks.py [hunk 1]
```diff
@@ -98,22 +98,22 @@ def end_pat(self):
 
 
 class TagIterator:
-    def __init__(self, data):
-        self.data = data
+    def __init__(self, text):
```
Comment (MichelleArk) [inline dbt_common/clients/_jinja_blocks.py:101]: while we're here and have context, would you mind adding some type annotations? lots to backfill in this repo!

Comment (peterallenwebb) [reply dbt_common/clients/_jinja_blocks.py:101]: sure.

Comment (peterallenwebb) [reply dbt_common/clients/_jinja_blocks.py]: Done!

**Review by MichelleArk — APPROVED** [review]

File: dbt_common/clients/_jinja_blocks.py (hunk 1 above)
Comment (MichelleArk) [reply dbt_common/clients/_jinja_blocks.py]: thank you thank you! ✨

**Pull request merged**